src/ringct/rctSigs.cpp

```diff
@@ -1749,4 +1750,52 @@ namespace rct {
       key mask;
       return decodeRctSimple(rv, sk, i, mask, hwdev);
     }
+
+    key getCommitment(const cryptonote::transaction &tx, const std::size_t output_idx) {
+      const bool miner_tx = cryptonote::is_coinbase(tx);
+      if (miner_tx || tx.version < 2)
+      {
+        CHECK_AND_ASSERT_THROW_MES(tx.vout.size() > output_idx, "unexpected size of vout");
+        return zeroCommitVartime(tx.vout[output_idx].amount);
+      }
+      CHECK_AND_ASSERT_THROW_MES(tx.rct_signatures.outPk.size() > output_idx, "unexpected size of outPk");
+      return tx.rct_signatures.outPk[output_idx].mask;
+    }
+
+    bool verPointsForTorsion(const std::vector<key> & pts) {
+      if (pts.empty())
+        return true;
+
+      tools::threadpool& tpool = tools::threadpool::getInstanceForCompute();
+      tools::threadpool::waiter waiter(tpool);
+
+      std::deque<bool> torsion_free;
```

Comment:
You could prevent the allocation of a `deque` by using a `std::atomic_flag` shared for all points.

Reply:
Solid idea. I think would rather keep `deque` for now and not try something new here just because that's how the rest of the file does it.